### ISSUE TITLE
How to train a Shared layer multi-task DNN

### ISSUE THREAD

**Issue by haoqi** [fchollet/keras#1441]
Hi guys,
I am trying to implement a DNN with shared hidden layer but with different output layer on the top towards different task related to different training data. Thus, I shuffle training data to get all different training data in one mini-bach.
Also, I need to modify the back propagation a little bit, when a training data is presented to the DNN trainer, only the shared hidden layers and corresponding top softmax layer are updated. How can I do this in Keras? It is kind of hard for me to handle this, hopefully you can give me some help.

For example, we have two language: English(lang1) and French(lang2), and want to obtain shared information of language and get different output towards different language.
Thanks a lot!!

![unnamed0](https://cloud.githubusercontent.com/assets/5073714/12219701/e77736d2-b703-11e5-94f9-a561203ae357.png)

**Comment by dasguptar** [comment]
You can have a look at the `Graph` container. It supports DAG structures like the one you are looking for.

**Comment by haoqi** [comment]
I had reviewed Graph class, and the sample example of Graph in Kerea.io. There are still some problems in my case. @fchollet 
lang1 and lang2 in my case can be treated as two kinds of training dataset, they have different training labels. Once I shuffled the data, it has mixed training data in one mini-batch. 
I want to ask if there are some solutions, like there is a ref towards each sample(not for a minibatch), which update the hidden layer parameters only depending on those valid labels error, like specific dropout towards samples.
 Thanks a lot!
